agent/plugins/modules/preset_config.py

```diff
@@ -0,0 +1,444 @@
+        provisioning_network, external_subnet_v4, external_subnet_v6 = "fd00:1101::0/64", "192.168.111.0/24", "fd2e:6f44:5dd8:c956::/120"
+        external_network = external_subnet_v6
+    else:
+        pass
+
+    if ip_stack == 'v4':
+        network_type = "OpenShiftSDN"
+        cluster_subnet_v4, cluster_host_prefix_v4, service_subnet_v4 = "10.128.0.0/14", "23", "172.30.0.0/16"
+        cluster_subnet_v6, cluster_host_prefix_v6, service_subnet_v6 = "", "", ""
+        cluster_network, cluster_host_prefix, service_network = cluster_subnet_v4, cluster_host_prefix_v4, service_subnet_v4
+    elif ip_stack == 'v6':
+        network_type = "OVNKubernetes"
+        cluster_subnet_v4, cluster_host_prefix_v4, service_subnet_v4 = "", "", ""
+        cluster_subnet_v6, cluster_host_prefix_v6, service_subnet_v6 = "fd01::/48", "64", "fd02::/112"
+        cluster_network, cluster_host_prefix, service_network = cluster_subnet_v6, cluster_host_prefix_v6, service_subnet_v6
+        mirror_images = True
+    elif ip_stack == 'v4v6':
+        network_type = "OVNKubernetes"
+        cluster_subnet_v4, cluster_host_prefix_v4, service_subnet_v4 = "10.128.0.0/14", "23", "172.30.0.0/16"
+        cluster_subnet_v6, cluster_host_prefix_v6, service_subnet_v6 = "fd01::/48", "64", "fd02::/112"
+        cluster_network, cluster_host_prefix, service_network = cluster_subnet_v4, cluster_host_prefix_v4, service_subnet_v4
+    else:
+        pass
+
+    if provisioning_network_profile == 'Disabled':
+        if host_ip_stack == 'v6':
+            provisioning_ip_subnet = external_subnet_v6
+        else:
+            provisioning_ip_subnet = external_subnet_v4
+
+    facts = {
+        'ip_stack': ip_stack,
+        'host_ip_stack': host_ip_stack,
+        'network_type': network_type,
+        'cluster_subnet_v4': cluster_subnet_v4,
+        'cluster_host_prefix_v4': cluster_host_prefix_v4,
+        'service_subnet_v4': service_subnet_v4,
+        'cluster_subnet_v6': cluster_subnet_v6,
+        'cluster_host_prefix_v6': cluster_host_prefix_v6,
+        'service_subnet_v6': service_subnet_v6,
+        'cluster_network': cluster_network,
+        'cluster_host_prefix': cluster_host_prefix,
+        'service_network': service_network,
+        'provisioning_network': provisioning_network,
+        'external_subnet_v4': external_subnet_v4,
+        'external_subnet_v6': external_subnet_v6,
+        'external_network': external_network,
+        'provisioning_network_profile': provisioning_network_profile,
+        'provisioning_ip_subnet': provisioning_ip_subnet,
+        'mirror_images': mirror_images,
+        'base_static_ip': '80'
+    }
+
+    return facts
+
+def generate_preset_agent(agent_static_ip_node0_only):
+
+    facts = {
+        'agent_static_ip_node0_only': agent_static_ip_node0_only
+    }
+
+    return facts
+
+# def generate_node_hostnames_for_static_ips(cluster_name, num_masters, num_workers, agent_static_ip_node0_only):
```

Comment:
This is leftover I'm debating whether it is needed or not. This information is pulled from the libvirt XML / ironic_nodes.json  instead of calculating it.

Reply:
I removed these code blocks.